File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = {0, 0};
        
        for(int i=3; i<brown+yellow; i++) {
            int x = (brown+yellow) / i;
            if (x >= i && ((i-2) * (x-2) == yellow)) {
                answer[0] = x;
                answer[1] = i;
                break;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
import math

def solution(brown, yellow):
    answer = [0, 0]
    
    by = brown + yellow

    for i in range(3, int(math.sqrt(by)) + 1):
        if by % i == 0 and (by // i - 2) * (i - 2) == yellow:
            return [by // i, i]